src/java.base/share/classes/jdk/internal/util/Exceptions.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2018, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.util;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.UnixDomainSocketAddress;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import sun.security.util.SecurityProperties;
+import jdk.internal.misc.VM;
+
+/**
+ * Contains static utility methods which can filter exception
+ * message strings for sensitive information.
+ *
+ * Code using this mechanism should use formatMsg()
+ * to generate a formatted (enhanced or restricted) string for exception
+ * messages.
+ *
+ * The methods above take variable numbers of SensitiveInfo objects
+ * as parameters which contain the text that may have to be filtered.
+ *
+ * The SensitiveInfo objects should be generated with one of the following:
+ *     public static SensitiveInfo filterSocketInfo(String s)
+ *     public static SensitiveInfo filterNonSocketInfo(String s)
+ *     public static SensitiveInfo filterJarName(String name)
+ *     public static SensitiveInfo filterUserName(String name)
+ */
+public final class Exceptions {
```

Comment:
maybe a refactor rename to convey some of the semantics of the class e.g. ExceptionMessageFilter

Reply:
I see what you mean, but the class does more than just filter messages. It also generates new Exception objects from existing ones. Given that its original name was `SocketExceptions` and all we've done here is generalize it to work with other exception types, I think `Exceptions` is a reasonable name.